src/trading/tradingSdk.test.ts

```diff
@@ -74,18 +74,100 @@ describe('TradingSdk', () => {
     it('When logs option is set to true, then should display logs', async () => {
       const logSpy = jest.spyOn(console, 'log')
 
-      const sdk = new TradingSdk(
+      const sdk = new TradingSdk(defaultTrader, { enableLogging: true, orderBookApi })
+
+      await sdk.getQuote(defaultOrderParams)
+
+      expect(logSpy.mock.calls[0][0]).toContain('[COW TRADING SDK]')
+    })
+  })
+
+  describe('Quote and post limit order', () => {
+    it('Quote result order amounts should match the quote result', async () => {
+      // This value is calculated in suggestSlippageBps.ts
+      const smartSlippage = 56
+
+      const sdk = new TradingSdk(defaultTrader, { orderBookApi })
+
+      // We requested a quote for defaultOrderParams
+      const quote = await sdk.getQuote(defaultOrderParams)
+
+      // orderToSign sell amount is quote.sellAmount - quote.feeAmount
+      expect(+quote.quoteResults.orderToSign.sellAmount).toEqual(
+        Number(quoteResponseMock.quote.sellAmount) + Number(quoteResponseMock.quote.feeAmount),
+      )
+
+      expect(quote.quoteResults.suggestedSlippageBps).toBe(smartSlippage)
+
+      // 397760000000000000000
+      const expectedBuyAmountFromQuote = amountMinusPercent(
+        Number(quoteResponseMock.quote.buyAmount),
+        smartSlippage / 100,
+      )
+
+      // orderToSign buy amount is quote.buyAmount - slippage
+      expect(+quote.quoteResults.orderToSign.buyAmount).toEqual(expectedBuyAmountFromQuote)
+    })
+
+    it('The final order should take partnerFee and slippage into account', async () => {
+      const slippageBips = 70
+      const partnerFeeBps = 30
+      const smartSlippage = 56
+
+      const sdk = new TradingSdk(defaultTrader, { orderBookApi })
+
+      const quote = await sdk.getQuote(defaultOrderParams)
+
+      // orderToSign buy amount is quote.buyAmount - slippage
+      // 397760000000000000000
+      const expectedBuyAmountFromQuote = amountMinusPercent(
+        Number(quoteResponseMock.quote.buyAmount),
+        smartSlippage / 100,
+      )
+
+      // Add partnerFee and slippage to the final order appData just before signing and sending it
+      const appDataParams: AppDataParams = {
+        metadata: {
+          quote: {
+            slippageBips,
+          },
+          partnerFee: {
+            volumeBps: partnerFeeBps,
+            recipient: '0xrecipient',
+          },
+        },
+      }
+
+      await sdk.postLimitOrder(
         {
-          chainId: SupportedChainId.GNOSIS_CHAIN,
-          appCode: 'test',
-          signer: '0xa43ccc40ff785560dab6cb0f13b399d050073e8a54114621362f69444e1421ca',
+          ...quote.quoteResults.tradeParameters,
+          ...quote.quoteResults.orderToSign,
         },
-        { enableLogging: true, orderBookApi }
+        { appData: appDataParams },
       )
```

Comment:
Don't we need to pass the `additionalParams: {networkCostsAmount: '1111' }` so it's available for the amounts calculations?

Here's how AAVE is doing it for reference https://github.com/aave/interface/blob/05df57b7e3ce78c147b6e36b5dbf124f2b80d47f/src/components/transactions/Switch/cowprotocol.helpers.ts#L217-L233
They don't have the quote obj by this point.

Reply:
No need to pass `networkCostsAmount` as an additional parameter, because we use `quote.quoteResults.orderToSign.sellAmount` for posting a limit order, and this amount already includes the network costs.

Reply:
We could do this, but the result will be the same:

<img width="1055" height="267" alt="image" src="https://github.com/user-attachments/assets/05fc7e67-8ee8-4734-9f05-593a094102a7" />